prog/collide.go

```diff
@@ -55,3 +58,46 @@ func AssignRandomAsync(origProg *Prog, rand *rand.Rand) *Prog {
 
 	return prog
 }
+
+// We append prog to itself, but let the second part only reference resource from the first one.
+// Then we execute all the duplicated calls simultaneously.
+// This somehow resembles the way the previous collide mode was implemented - a program was executed
+// normally and then one more time again, while keeping resource values from the first execution and
+// not waiting until every other call finishes.
+func DoubleExecCollide(origProg *Prog, rand *rand.Rand) (*Prog, error) {
+	if len(origProg.Calls)*2 > MaxCalls {
+		return nil, fmt.Errorf("the prog is too big for the DoubleExecCollide transformation")
+	}
+	prog := origProg.Clone()
+	oldToNew := make(map[*ResultArg]*ResultArg)
+	dupCalls := cloneCalls(prog.Calls, oldToNew)
```

Comment:
We replace old args to new while cloning and then replace them back, right?
I wonder if we could not replace args during cloning in the first place? Maybe if we pass nil oldToNew map to cloneCalls, it will not replace old to new. Would it work?

Reply:
> We replace old args to new while cloning and then replace them back, right?

In fact, yes.

> Maybe if we pass nil oldToNew map to cloneCalls, it will not replace old to new.

Adding an extra `if` to the code working with the `newargs` map inside the `clone` function should be enough to make it work. The interface to the `cloneCalls` method would become somewhat less intuitive, though.

Reply:
> The interface to the cloneCalls method would become somewhat less intuitive, though.

You can make it an additional bool flag with a descriptive name.